### PR TITLE
Create Feature image to base64

### PR DESCRIPTION
**Pull request by that-ambuj**
Hi @souvikinator, I have built the image link to base64 feature as we discussed in #60 with unit tests too but there are a few issues that I have faced:

- Base64 format is not widely supported by many markdown parsers and renderers. I have tried Hugo, Github and markdown-it markdown parser.
- The conversion performed using this features is very tricky and should be used carefully as some links ~return a javascript file~ redirect(like https://picsum.photos) the browser(not sure how this workes in node js) run instead of a raw image bytes; in that case the javascript code gets encoded to base64. PS: this issue is irrelevant as redirects are followed by node-fetch but there still might be some uncertainties.
- People usually prefer images to be downloaded as files or permanently hosted in a storage bucket when using this library instead of base64, because of the aforementioned lack of support by popular markdown parsers and static-site generators.

I think it is understandable how base64 does not work with major markdown parsers because of how tricky and error prone this method of rendering images is(in case of point 2, it would produce a corrupted image, and the data is not guaranteed to be from an actual image) and the amount of challenge this feature produces is not worth the effort and a very small community trying to use this feature. Instead, I propose you that we work on temporary s3 links and already unsupported base64 data being emitted by this library.

### CODE REVIEW

**Comment by souvikinator** [comment]
> the javascript code gets encoded to base64

That doesn't sound good and may lead to certain vulnerabilities.  However I agree with your proposal,  we can make it Notion's temporary s3 links exclusive and parse other links to regular markdown images syntax.

File: src/notion-to-md.ts [hunk 1]
```diff
@@ -20,9 +19,11 @@ import { getBlockChildren } from "./utils/notion";
 export class NotionToMarkdown {
   private notionClient: Client;
   private customTransformers: Record<string, CustomTransformer>;
+  private convertImagesToBase64: boolean;
```
Comment (souvikinator) [inline src/notion-to-md.ts:22]: can you move this under the config option as per the latest changes in the master?

File: src/types/index.ts [hunk 2]
```diff
@@ -16,6 +16,7 @@ export type TextRequest = string;
 
 export interface NotionToMarkdownOptions {
   notionClient: Client;
+  convertImagesToBase64?: boolean;
```
Comment (souvikinator) [inline src/types/index.ts:19]: please sync with the latest changes in the master.

https://github.com/souvikinator/notion-to-md/blob/5cd0b5e1d596d7d5ddfd4698582b196c5912c636/src/types/index.ts#L22

File: src/notion-to-md.ts (hunk 1 above)
Comment (that-ambuj) [reply src/notion-to-md.ts:22]: yeah that sounds good.

**Comment by that-ambuj** [comment]
Hi, the last two commits fix the base64 not being rendered by the markdown parsers and base64 images are now working. I'll also add a similar fix for the base64 data(coming from notion) that does not need conversion. To my surprise, the redirected images are now working as I've tested the base64 conversion with a picsum.photos link. I don't think there'll be any javascript embedded or executed in the base64 image data as node-fetch handles redirects very well.

**Comment by that-ambuj** [comment]
Hey @souvikinator, I've merged the latest changes in the feature branch. There are no merge conflicts, everything is tested and ready to merge!

**Comment by that-ambuj** [comment]
For reference, there is a page on MDN about everything you need to know about base64 images and `data:` prefixed urls: https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/Data_URLs . Keep in mind this is modern feature and the support for this is listed at the bottom of the aforementioned page.

**Comment by souvikinator** [comment]
Awesome! Thanks for the contribution. This really is an amazing feature. Should be live in upcoming major update.

**Comment by that-ambuj** [comment]
It was my pleasure to work on this project! I'm looking forward to adding many more features to this project which you'll see in the future and issues and PRs that I open.

Please also update the issues mentioning this PR and #60.

**Comment by souvikinator** [comment]
Hi again! Your past feedback on notion-to-md has been valuable. 

We're now working on **v4** with major improvements to address various use cases better.  I'd love your insights on this. Please checkout the GitHub discussion #112   to discuss or share your feedback? Your feedback would be really valuable in shaping v4's development.

**Pull request merged**